src/java.management/share/classes/com/sun/jmx/mbeanserver/PerInterface.java

```diff
@@ -108,8 +108,7 @@ Object invoke(Object resource, String operation, Object[] params,
         final List<MethodAndSig> list = ops.get(operation);
         if (list == null) {
             final String msg = "No such operation: " + operation;
-            return noSuchMethod(msg, resource, operation, params, signature,
-                                cookie);
+            throw new ReflectionException(new NoSuchMethodException(operation + sigString(signature)), msg);
```

Comment:
I think this is much better, but I still think it's odd that `operation` is in both exceptions (after your changes it's a lot more clear that this is the case), but I'm okay with that for the sake of not risking any compatibility issues.